templates/vue/src/store/mutations.js

```diff
@@ -116,18 +117,31 @@ export function deleteLink(state, { source, target }) {
 }
 
 // quizzes
-export function completeQuestion(state, { nodeId, questionId }) {
+export function completeQuestion(state, { nodeId, questionId, answerType, answer }) {
   const node = getters.getNode(state)(nodeId)
-  const question = node.quiz.find(question => question.id === questionId)
+
+  const question = node.typeData.activity.questions.find(
+    question => question.id === questionId
+  )
+
   question.completed = true
-}
 
-export function updateEntry(state, { answerType, entry, nodeId, questionId }) {
-  const node = getters.getNode(state)(nodeId)
-  const question = node.quiz.find(question => question.id === questionId)
-  const entries = question.entries || {}
-  entries[answerType] = Object.values(entry)[0]
-  question.entries = entries
+  if (
+    state.userAnswers[nodeId] === undefined ||
+    state.userAnswers[nodeId].activity === undefined
+  ) {
+    state.userAnswers[nodeId] = {}
+    state.userAnswers[nodeId].activity = {}
+  }
+  if (state.userAnswers[nodeId].activity[questionId] === undefined) {
+    state.userAnswers[nodeId].activity[questionId] = {}
+    state.userAnswers[nodeId].activity[questionId].answers = {}
+  }
+  if (typeof state.userAnswers[nodeId].activity[questionId].answers === "string") {
+    state.userAnswers[nodeId].activity[questionId].answers = {}
+  }
```

Comment:
Again, not sure why this would be a string in the first place? If this is being returned from the backend, the backend should not return a string and just return an empty object in this case so the frontend doesn't have to get involved in this.